auth/src/user.rs

```diff
@@ -337,19 +403,33 @@ impl FromRow<'_, PgRow> for Subscription {
                 .ok()
                 .and_then(|inner| SubscriptionId::from_str(inner).ok())
                 .unwrap(),
-            r#type: ShuttleSubscriptionType::from_str(row.try_get("type").unwrap()).map_err(
-                |err| sqlx::Error::ColumnDecode {
+            r#type: models::user::SubscriptionType::from_str(row.try_get("type").unwrap())
+                .map_err(|err| sqlx::Error::ColumnDecode {
                     index: "type".to_string(),
                     source: Box::new(std::io::Error::new(ErrorKind::Other, err.to_string())),
-                },
-            )?,
+                })?,
             quantity: row.try_get("quantity").unwrap(),
             created_at: row.try_get("created_at").unwrap(),
             updated_at: row.try_get("updated_at").unwrap(),
         })
     }
 }
 
+impl From<User> for Limits {
+    fn from(user: User) -> Self {
+        let mut limit: Limits = user.account_tier.into();
+
+        limit.rds_quota = user
+            .subscriptions
+            .iter()
+            .find(|sub| matches!(sub.r#type, models::user::SubscriptionType::Rds))
+            .map(|sub| sub.quantity as u32)
+            .unwrap_or(0);
+
+        limit
+    }
+}
```

Comment:
I'm thinking that we need to sync the rds subscription with stripe before we do this. This way, if a user has not paid their subscription, they will not be permitted to provision new RDS instances.

We already sync the account tier when we fetch the user, perhaps we can do the same for RDS? The current sync call is a bit hidden, though, perhaps there is a clearer way. The current sync logic is also a bit complicated.

Reply:
We'd then also need to mock the call to Stripe in the test.

Reply:
I was considering this. But since we decided in the TR to leave the sync for a later time I decided to just trust the DB (cache) for now

Reply:
Ah, I interpreted that decision as we won't refactor to a cron-based sync yet, but we could still sync on actions that create a JWT with an RDS quota. They would be limited to their quota, however, so I suppose the worst case is that if they delete an instance to go under their quota, they can still create a new one indefinitely after they stop paying their subscription. I'm not sure what the timeline is for the subscription sync refactor, but if it's not too distant in the future, I think we can leave it as is for now.